[re2] Port is broken due to invalid version date (2025-06-26)

1 ### Package
    2
    3 re2
    4
    5 ### Vcpkg version
    6
    7 Latest as of 2025-07-17
    8
    9 ### Triplet
   10
   11 x64-windows-static (and likely others)
   12
   13 ### What happened?
   14
   15 When attempting to install `grpc:x64-windows-static`, the build fails because its dependency `re2` cannot be downloaded.
   16
   17 The root cause is that the `re2` port points to an invalid version.
   18
   19 1.  The file `ports/re2/vcpkg.json` specifies a `version-date` of `2025-06-26`.
   20 2.  This causes `vcpkg_from_github` in `ports/re2/portfile.cmake` to generate a non-existent download URL:
      `https://github.com/google/re2/archive/2025-06-26.tar.gz`.
   21 3.  This URL results in a 404 error, causing the build to fail.
   22
   23 The latest available release tag on the `google/re2` GitHub repository is `2024-07-02`. The port seems to contain a typo in the
      year.
   24
   25 ### How to reproduce
   26
   27 1.  Make sure vcpkg is updated to the latest version.
   28 2.  Run `vcpkg install re2:x64-windows-static` or any package that depends on it, like `vcpkg install grpc:x64-windows-static`.
   29 3.  The build will fail with the following error:
  CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:136 (message):
    Download failed, halting portfile.
  Call Stack (most recent call first):
    scripts/cmake/vcpkg_from_github.cmake:120 (vcpkg_download_distfile)
    ports/re2/portfile.cmake:1 (vcpkg_from_github)
    scripts/ports.cmake:206 (include)

  error: building re2:x64-windows failed with: BUILD_FAILED

   1
   2 ### Expected behavior
   3
   4 The `re2` port should point to a valid, existing release version and download successfully. The `version-date` in `vcpkg.json`
     should likely be `2024-07-02` (or another recent, valid tag), and the SHA512 in `portfile.cmake` should be updated to match.

--- a/ports/re2/portfile.cmake
+++ b/ports/re2/portfile.cmake
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO google/re2
-    REF "${VERSION}"
-    SHA512 5336c9aa1dcf528290bfae08f277f2d6d9a590cca3ece362b3f37d13192a13cd8580afe1919bdff137de22cfdb4254269cbcc4a7835d65962278d68ff2886d53
+    REF 8ea5841693c6c0b837c6ed2189217e8f8d6fee9c
+    SHA512 547ba362da00457fd0687ce1c45ba74ef5426c82abab66b9e05c13401730195c928ec301b5d6bf65216d6e595dab2d67f486e83f0a8fd73a07aab9d4ad138c5f
     HEAD_REF master
 )
 
@@ -13,11 +13,10 @@ vcpkg_cmake_configure(
 )
 
 vcpkg_cmake_install()
+vcpkg_copy_pdbs()
 vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/${PORT}")
 vcpkg_fixup_pkgconfig()
 
-vcpkg_copy_pdbs()
-
-# Handle copyright
-vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

--- a/ports/re2/vcpkg.json
+++ b/ports/re2/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "re2",
   "version-date": "2025-06-26",
+  "port-version": 1,
   "description": "RE2 is a fast, safe, thread-friendly alternative to backtracking regular expression engines like those used in PCRE, Perl, and Python. It is a C++ library.",
   "homepage": "https://github.com/google/re2",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -8222,7 +8222,7 @@
     },
     "re2": {
       "baseline": "2025-06-26",
-      "port-version": 0
+      "port-version": 1
     },
     "reactiveplusplus": {
       "baseline": "2.1.1",

--- a/versions/r-/re2.json
+++ b/versions/r-/re2.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "6f2f4a79f81187bd13e71c6dea2db2d253ee4e51",
+      "version-date": "2025-06-26",
+      "port-version": 1
+    },
+    {
       "git-tree": "2ab7d8d8a81010aa0d531970b9680091df510725",
       "version-date": "2025-06-26",
       "port-version": 0

